### PR TITLE
[android][expo-network] migrate to new modules API

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Migrate native modules to use new modules API

# How
Followed the usual steps involved in migrating a module

# Test Plan
Tested in the bare expo app. All tests pass as before

### CODE REVIEW

File: packages/expo-network/android/src/main/java/expo/modules/network/NetworkModule.kt [hunk 1]
```diff
@@ -1,31 +1,80 @@
...
+
+          result.apply {
+            putString("type", connectionType?.value ?: NetworkStateType.NONE.value)
+            putBoolean("isInternetReachable", isInternetReachable)
+            putBoolean("isConnected", connectionType != null && connectionType.isDefined)
+          }
+          promise.resolve(result)
+        }
+      } catch (e: Exception) {
+        throw NetworkAccessException()
```
Comment (Kudo) [inline packages/expo-network/android/src/main/java/expo/modules/network/NetworkModule.kt:66]: would be better to pass the exception into the `NetworkAccessException`. the [`CodedException` ](https://github.com/expo/expo/blob/77aca4d3c9a8c43d058c5e272a3be99d6ef20352/packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/exception/CodedException.kt#L16) has an optional `cause` parameter for this.

**Review by Kudo — COMMENTED** [review]
thanks for the super awesome expo modules api migration again 🔥
i'm just leaving some nit comments here. most changes look great

**Comment by alanjhughes** [comment]
Thank you for your review @Kudo! I've made those changes

**Review by Kudo — APPROVED** [review]
awesome! thanks for the updates 👏  let's land it 🚀

**Comment by alanjhughes** [comment]
Thanks @Kudo! 👍

**Pull request merged**